### PR TITLE
Add ElfFile::LoadEhOrDebugFrameEntriesAsSymbols

### PR DESCRIPTION
**Pull request by dpallotti**
This is the counterpart of `CoffFile::LoadExceptionTableEntriesAsSymbols` (https://github.com/google/orbit/pull/4178).
It uses `.debug_frame` if available and falls back to `.eh_frame`, which is usually the only one present.
Again, it assigns an arbitrary name to the resulting function symbols.

We need some ugly and specific handling of `.eh_frame` entries as returned by LLVM because of a bug only fixed in LLVM 13.

Bug: http://b/245680119

Test:
- Unit tests.
- As part of the larger symbol loading fallback prototype.

### CODE REVIEW

File: src/ObjectUtils/ElfFileTest.cpp [hunk 1]
```diff
@@ -77,19 +96,83 @@ TEST(ElfFile, LoadSymbolsFromDynsym) {
...
+
+  const auto symbols_result = elf_file->LoadEhOrDebugFrameEntriesAsSymbols();
+  ASSERT_THAT(symbols_result, HasNoError());
+
+  std::vector<SymbolInfo> symbol_infos(symbols_result.value().symbol_infos().begin(),
+                                       symbols_result.value().symbol_infos().end());
+  // These can be obtained with `objdump hello_world_elf --dwarf=frames` looking at the FDE entries.
+  EXPECT_THAT(
+      symbol_infos,
+      testing::ElementsAre(SymbolInfoEq("[function@0x1050]", 0x1050, 43),  // `_start`
```
Comment (florian-kuebler) [inline src/ObjectUtils/ElfFileTest.cpp:152]: Nice to test actually ALL symbols here.

File: src/ObjectUtils/ElfFile.cpp [hunk 2]
```diff
@@ -416,6 +418,76 @@ ErrorMessageOr<ModuleSymbols> ElfFileImpl<ElfT>::LoadSymbolsFromDynsym() {
...
+  constexpr const char* kErrorMessage =
+      "Unable to load unwind info ranges from the .debug_frame or the .eh_frame section.";
+  if (dwarf_context == nullptr) {
+    return ErrorMessage{kErrorMessage};
+  }
+
+  const llvm::DWARFDebugFrame* debug_or_eh_frame = nullptr;
+  bool is_eh_frame = false;
+
+  if (llvm::Expected<const llvm::DWARFDebugFrame*> debug_frame = dwarf_context->getDebugFrame();
```
Comment (florian-kuebler) [inline src/ObjectUtils/ElfFile.cpp:435]: I think this deserves a comment (similar to the one in libunwindstack) that we prioritize debug_frame over eh_frame.

**Review by florian-kuebler — APPROVED** [review]
LGTM. Only minor comments.

**Review by dpallotti — COMMENTED** [review]
I thought I had sent the reply? Maybe not.

**Review by florian-kuebler — APPROVED** [review]

**Pull request merged**